powerscale/helper/synciq_rule.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright (c) 2024 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"context"
+	powerscale "dell/powerscale-go-client"
+	"errors"
+	"terraform-provider-powerscale/client"
+	"terraform-provider-powerscale/powerscale/models"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// GetAllSyncIQRules retrieve the cluster information.
+func GetAllSyncIQRules(ctx context.Context, client *client.Client) (*powerscale.V3SyncRules, error) {
+	resp, _, err := client.PscaleOpenAPIClient.SyncApi.ListSyncv3SyncRules(context.Background()).Execute()
```

Comment:
This API call may not return all rules. Please check https://github.com/dell/terraform-provider-powerscale/pull/188/commits/04f42788c21d7609f2465b43b4f6472f31a4fc6d to see how to get all rules in case of paginated results.

Reply:
updated